src/Core/GoAspectContainer.php

```diff
@@ -88,19 +89,35 @@ public function __construct()
             );
         });
 
-        $this->share('aspect.annotation.reader', function(Container $container) {
+        $this->share('aspect.annotation.cache', function(Container $container) {
             $options = $container->get('kernel.options');
-            $reader  = new AnnotationReader();
+
             if (!empty($options['annotationCache'])) {
-                $reader = new CachedReader(
-                    $reader,
-                    $options['annotationCache'],
-                    $options['debug']
+                return $options['annotationCache'];
+            }
+
+            if (!empty($options['cacheDir'])) {
+                return new DoctrineCache\FilesystemCache(
+                    $options['cacheDir'] . DIRECTORY_SEPARATOR . '_annotations' . DIRECTORY_SEPARATOR,
+                    '.annotations.cache',
+                    0777 & (~$options['cacheFileMode'])
                 );
             }
 
-            return $reader;
+            return new DoctrineCache\ArrayCache();
```

Comment:
This can occur when running unit tests, so I fallback to ArrayCache in order for tests to pass.